### ISSUE TITLE
Can the database stop the query statement being executed to cancel the blocking

### ISSUE THREAD

**Issue by Sudingyi** [mattn/go-oci8#440]

**Comment by mattn** [comment]
Yes. Use the context variants (QueryContext / ExecContext / PingContext) with a cancelable context. When the context is canceled or times out while a statement is executing, the driver calls OCIBreak to interrupt the running OCI call, the statement is aborted server-side, and the call returns. See TestContextTimeoutBreak in oci8_sql_test.go for an example.
